Add API docs for new withStorybook wrapper (#610)
storybookjs/react-native: README.md

- [Heres an approach for react native cli](https://dev.to/dannyhw/multiple-entry-points-for-react-native-storybook-4dkp)
- [Heres an article about how you can do it in expo](https://dev.to/dannyhw/how-to-swap-between-react-native-storybook-and-your-app-p3o)

## withStorybook wrapper

> `withStorybook` is a wrapper function for your metro config. It accepts your existing [Metro config](https://metrobundler.dev/docs/configuration), and supports a variety of options for how Storybook should be started and configured.

Reviewer: ```suggestion
`withStorybook` is a wrapper function to extend your [Metro config](https://metrobundler.dev/docs/configuration) for Storybook. It accepts your existing Metro config and an object of options for how Storybook should be started and configured.
```

```js
// metro.config.js
const { getDefaultConfig } = require('expo/metro-config');
const withStorybook = require('@storybook/react-native/metro/withStorybook');

const defaultConfig = getDefaultConfig(__dirname);

module.exports = withStorybook(defaultConfig, {
  enabled: true,
  // See API section below for available options
});
```

### Options

#### enabled

Type: `boolean`, default: `true`

> Whether or not to show Storybook in the UI. Can by dynamically configured by a `process.env` variable.

Reviewer 2: ```suggestion
Configures the UI to display Storybook. It can be dynamically configured via `process.env` variable.
```

Reviewer 3: hmm actually this is kinda both inaccurate, the option will actually just change if the metro options are applied or not, showing the ui or not is controlled by the user in the main entrypoint

the intention is just to not have configuration options specific for storybook mixing with config for the main application if running a setup where you switch between the two.

Reviewer: ````suggestion
Determines whether the options specified are applied to the Metro config. This can be useful for project setups that use Metro both with and without Storybook and need to conditionally apply the options. In this example, it is made conditional on an environment variable:

```js
// metro.config.js
const { getDefaultConfig } = require('expo/metro-config');
const withStorybook = require('@storybook/react-native/metro/withStorybook');

const defaultConfig = getDefaultConfig(__dirname);

module.exports = withStorybook(defaultConfig, {
  enabled: process.env.WITH_STORYBOOK,
  // ... other options
});
```
````

Reviewer 2: @dannyhw, are you ok with something of this nature?

```suggestion
Allows Storybook's configuration to be applied to the project by default without overriding the existing configuration when switching between environments.
```

#### useJs

Type: `boolean`, default: `false`

> Whether or not to generate the `.storybook/storybook.requires` file in Javascript or TypeScript.

Reviewer 2: @dannyhw correct me if I'm wrong, but the file (i.e., `.storybook/storybook.requires` is created with TypeScript by default right? 
```suggestion
Generates the `.storybook/storybook.requires` file in JavaScript—defaults to TypeScript.
```

Reviewer 3: yes its typescript by default

Reviewer: ```suggestion
Generates the `.storybook/storybook.requires` file in JavaScript instead of TypeScript.
```

#### configPath

Type: `string`, default: `path.resolve(process.cwd(), './.storybook')`

> The location of your Storybook configuration directory, including `main.ts` and other project-related files.

Reviewer: ```suggestion
The location of your Storybook configuration directory, which includes `main.ts` and other project-related files.
```

> The location of your Storybook configuration directory with `main.ts`, and o[tionally `preview.tsx` and other configuration files.

Reviewer 2: ```suggestion
The location of your Storybook configuration directory, including `main.ts` and other project-related files.
```

### websockets

Type: `{ host: string?, port: number? }`, default: `undefined`

> Create a websocket server on startup, if specified. This allows, for example, [React Native Web Storybook](https://github.com/storybookjs/addon-react-native-web) to control RN Storybook from a web browser.

Reviewer 3: Technically this isn't true since the web one needs something like addon-react-native-server, its more accurate to say something like:

```suggestion
Create a websocket server on startup, if specified. This allows you to sync up multiple devices to show the same story and state of args/controls
```

The video I created before was done all with ondevice ui

Reviewer 3: its kinda like you either use addon-react-native-server and use the web to control everything or you use this option in withStorybook, i.e one or the other not both

Reviewer 2: @dannyhw would something like this work for you:
```suggestion
If specified, create a WebSocket server on startup. This allows you to sync up multiple devices to show the same story and [arg](https://storybook.js.org/docs/writing-stories/args) values connected to the story in the UI.
```

> Create a websocket server on startup, if specified. This allows, for example, [React Native Web Storybook](https://github.com/storybookjs/addon-react-native-web) to control RN Storybook from a web browser.

Reviewer 2: ```suggestion
If specified, create a WebSocket server on startup. This allows, for example, [React Native Web Storybook](https://github.com/storybookjs/addon-react-native-web) to control Storybook's React Native from a web browser.
```

### websockets.host

Type: `string`, default: `'localhost'`

> The host on which to run the websocket, if specified.

Reviewer 2: ```suggestion
The host on which to run the WebSocket, if specified.
```

> ### websockets.host

Reviewer 3: ```suggestion
### websockets.port
```

> Type: `number`, default: `777`

Reviewer 3: default is 7007 currently

Author: oops thanks. super jet lagged

> The port on which to run the websocket, if specified.

Reviewer 2: ```suggestion
The port on which to run the WebSocket, if specified.
```
